compass/ocean/tests/soma/forward.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+
+from compass.step import Step
+from compass.model import partition, run_model
+from compass.ocean.particles import build_particle_simple
+
+
+class Forward(Step):
+    """
+    A step for performing forward MPAS-Ocean runs as part of SOMA test cases.
+
+    Attributes
+    ----------
+    resolution : str
+        The resolution of the test case
+
+    with_particles : bool
+        Whether to run with Lagrangian particles
+    """
+    def __init__(self, test_case, resolution, name='forward', subdir=None,
+                 with_particles=False):
+        """
+        Create a new test case
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        resolution : str
+            The resolution of the test case
+
+        name : str
+            the name of the test case
+
+        subdir : str, optional
+            the subdirectory for the step.  The default is ``name``
+
+        with_particles : bool, optional
+            Whether to run with Lagrangian particles
+        """
+        self.resolution = resolution
+        self.with_particles = with_particles
+        res_params = {'32km': {'cores': 4,
+                               'min_cores': 1,
+                               'dt': "'00:10:00'",
+                               'btr_dt': "'0000_00:00:25'",
+                               'mom_del4': "2.0e11"},
+                      '16km': {'cores': 4,
+                               'min_cores': 1,
+                               'dt': "'00:10:00'",
+                               'btr_dt': "'0000_00:00:25'",
+                               'mom_del4': "2.0e10 "},
+                      '8km': {'cores': 4,
+                              'min_cores': 1,
+                              'dt': "'00:10:00'",
+                              'btr_dt': "'0000_00:00:15'",
+                              'mom_del4': "2.0e9"},
+                      '4km': {'cores': 4,
+                              'min_cores': 1,
+                              'dt': "'00:06:00'",
+                              'btr_dt': "'0000_00:00:10'",
+                              'mom_del4': "4.0e8"}}
```

Comment:
We need to modify these.  Based on ~300 cells for the target number of cores and ~3000 for the minimum, we would have:
```suggestion
        res_params = {'32km': {'cores': 25,
                               'min_cores': 3,
                               'dt': "'00:10:00'",
                               'btr_dt': "'0000_00:00:25'",
                               'mom_del4': "2.0e11"},
                      '16km': {'cores': 100,
                               'min_cores': 10,
                               'dt': "'00:10:00'",
                               'btr_dt': "'0000_00:00:25'",
                               'mom_del4': "2.0e10 "},
                      '8km': {'cores': 400,
                              'min_cores': 40,
                              'dt': "'00:10:00'",
                              'btr_dt': "'0000_00:00:15'",
                              'mom_del4': "2.0e9"},
                      '4km': {'cores': 1600,
                              'min_cores': 160,
                              'dt': "'00:06:00'",
                              'btr_dt': "'0000_00:00:10'",
                              'mom_del4': "4.0e8"}}
```

Reply:
These should also be made config options instead of just hard coded.

Reply:
It would make sense to have `dt` be approximately proportional to the grid spacing.